### DISCUSSION TITLE
Connection lost

### DISCUSSION THREAD

**Discussion by Jlauper** [jpcornil-git/HA-sma#23] [Q&A]
HA lost connection after 4.5 day operation:
Screenshot (https://github.com/user-attachments/assets/fcef1175-6482-49f0-9d97-bbe388561f89)
Reboot of HA did not change anything.

I removed one Webbox and could not connect to this webbox anymore:
Error: RPC  rpc request timed out

Some Datapoints still there from 1 Box but Error in HA:
Screenshot (https://github.com/user-attachments/assets/f9a1b9d7-8205-41b7-a7b1-f472bcaad295)

The Log shows RPC request timed out to every box.

**Comment by jpcornil-git** [comment]
This seems to be a network issue, can you ping your webbox with the configured ip adresses from your HA instance ?

May be the webbox ip address changed (it should be configured as a static address or the dhcp server should assign a fixed address to it)

**Comment by Jlauper** [comment]
Webboxes have fixed IP.
I configured a ping sensor in HA. Every webbox is reporting connected.

**Comment by jpcornil-git** [comment]
Could you activate debug logs (Settings->Integrations-> SMA Webbox, bottom of the left card) and attach logs (Settings->System->Logs->Download full logs) ?

Did you try to connect with sma_webbox.py as well ?

**Comment by Jlauper** [comment]
sma_webbox.py reports [Errno 98] Adress in use
I get this message for all 3 webboxes, but there is just 1 box configured in SMA Webbox
ping is always possible to all 3 webboxes.

**Comment by Jlauper** [comment]
\[\e]0;\u@\h:\w\a\]\W $ python3 /homeassistant/custom_components/sma_webbox/sma_webbox.py 192.168.10.169
INFO:__main__:Starting SMA Webbox component
ERROR:__main__:Unable to connect to 192.168.10.169:34268 ([Errno 98] Address in use)
\[\e]0;\u@\h:\w\a\]\W $ python3 /homeassistant/custom_components/sma_webbox/sma_webbox.py 192.168.10.168
INFO:__main__:Starting SMA Webbox component
ERROR:__main__:Unable to connect to 192.168.10.168:34268 ([Errno 98] Address in use)
\[\e]0;\u@\h:\w\a\]\W $ python3 /homeassistant/custom_components/sma_webbox/sma_webbox.py 192.168.10.167
INFO:__main__:Starting SMA Webbox component
ERROR:__main__:Unable to connect to 192.168.10.167:34268 ([Errno 98] Address in use)

Webbox .168 is configured and running

**Comment by jpcornil-git** [comment]
On what platform are you running HA ?
This seems to be related to the Operating System not accepting new connections on port 34268 for some reasons.
I guess that instead of restarting HomeAssistant, you restart the main OS (if you are running HA within a containers, restart the OS running docker not the container), it will be fine ?

**Comment by Jlauper** [comment]
HA runs on a VM. I made an new install of SMA Webbox. Now everything works fine. 
I will see how it works in the next few days.

**Comment by jpcornil-git** [comment]
Is the host running the VM on a windows host (SO_REUSE_PORT socket option not well supported by Windows but required here because the udp port is fixed for all webboxes).

I'm running it since a few days before its release without issues up to now (HA running in a docker container on a Linux host but with a single webbox).

**Comment by Jlauper** [comment]
Yesterday I updated homeassistant. Now I get the RPC error message again. One webbox delivers data occasionally, the other two are no longer accessible. The RPC timeout error is reported in the LOG file.
Restarting the VM and Homeassistant did not change anything.
After restart of the Server Host all connections are lost.
The VM is running on Unraid (Alpine Linux)

**Comment by jpcornil-git** [comment]
RPC timeout indicates that no messages is received from the webbox, either because it hasn't been sent or received.

Connectivity issue could be related to:
1. python asyncio networking stack but when you restart HA, it restarts as well and reset issues if any
2. VM/docker networking stack, same as above when you restart the VM/docker (not the image/container but the VM/dockerd), issues related to the VM should dissappear
3. Host networking stack, same as above when you restart the host, issues related to the host should dissappear.
4. Your network but unlikely unless something wierd happens with mac/ip address assignations.
5. Webbox but unlikely as well unless multiple devices try to access it simultaneously/a 2nd instance is running.

Most likely issue is related to 2 or 3.

What do you mean by "restart of the server host" ? (reboot of the machine running your VM ... if this is the case, then it could be 4 or 5 but that would surprise me)

Wrt 5., restarting a webbox doesn't help I suppose ?

**Comment by jpcornil-git** [comment]
Where you able to recover the system using one of these steps ?

**Comment by Jlauper** [comment]
The problem can only be solved by reinstalling the Webbox component.
All Webboxes will immediately function normally again.

**Comment by jpcornil-git** [comment]
Just to make sure I understand; you mean that even if you power down everything and power up again, it doesn't recover and that to effectively recover, you have to remove the webbox integration and add it again to HA ?

If it is the case, it means that something wrong is stored in one of the HA registry (.storage/config_entries, .storage/device_registry or .storage/.entity_registry).

The trigger for the issue is an update of  homeassistant to a new version ? (I'll try to reproduce)

**Comment by Jlauper** [comment]
It's exactly like that.

**Comment by Jlauper** [comment]
After installing a webrtc camera integration with HACS, the connection to all webboxes is interrupted.
The webrtc camera integration has been uninstalled.
After reinstalling the SMA Webbox, everything works again.

**Comment by jpcornil-git** [comment]
Very strange that two integrations interfere with each other's (no shared/global objects apriori) unless webrtc is using the same port as SMA webboxes but that would be surprising. I'll have a look at that integration/see what it does.

Anything to see out of the logs from webbox or webrtc ?

**Comment by Jlauper** [comment]
The connection is broken again. At the same time, one webbox failed completely, one is still delivering partial data. One box is still working.
Unfortunately, I can't remember what could have caused the outage.

**Comment by jpcornil-git** [comment]
I just upgrade my own HA version to 11.2 to make sure and everything still work. (and not a single issue with the latest version of the integration either

When a box is failing, is sma_webbox.py working from a different machine ? if yes, from the same machine (host not in the container) ? if yes, from within HA container ?

**Comment by Jlauper** [comment]
After a power failure, the connection to 2 boxes was interrupted. Restarting Homeassistant did not help. I had to remove the 3 Webbox entities and restart the HA. I was then able to add the boxes again without any errors. The HA runs on a VM

**Comment by jpcornil-git** [comment]
A few more questions:
- When you restart HW, do you restart the VM or just HA ?
- Are you configuring this integration from configuration.yaml or using the user interface ?
- Could you upload the 3 lines corresponding to your webboxes from the .homeassistant/.storage/core.config_entries file (search for sma_webbox) ?
- Do you have the issue if you just restart HA ?

Something is "sticky" somewhere but I don't undertstand yet where ...

**Comment by Jlauper** [comment]
Normally i do not restart the VM just HA
A restart of Homeassistant breaks the connection to 2 of the 3 webboxes. Ping is always working to all of the boxes.

I am using the userinterface for configuration.

i can find the following 3 lines:

`  {"created_at":"2024-11-25T12:18:49.706444+00:00","data":{"ip_address":"192.168.10.169","port":34268,"scan_interval":30},"disabled_by":null,"discovery_keys":{},"domain":"sma_webbox","entry_id":"01JDHM56QAJVVKJEVK618X7QAS","minor_version":1,"modified_at":"2024-11-25T12:18:49.706454+00:00","options":{},"pref_disable_new>
      {"created_at":"2024-11-25T12:20:36.524722+00:00","data":{"ip_address":"192.168.10.167","port":34268,"scan_interval":30},"disabled_by":null,"discovery_keys":{},"domain":"sma_webbox","entry_id":"01JDHM8F1CZKR32PFTFPQFWN6M","minor_version":1,"modified_at":"2024-11-25T12:20:36.524726+00:00","options":{},"pref_disable_new>
      {"created_at":"2024-11-25T12:20:50.295557+00:00","data":{"ip_address":"192.168.10.168","port":34268,"scan_interval":30},"disabled_by":null,"discovery_keys":{},"domain":"sma_webbox","entry_id":"01JDHM8WFQ1A9BAYDH8HMQFJST","minor_version":1,"modified_at":"2024-11-25T12:20:50.295578+00:00","options":{},"pref_disable_new> `

**Comment by jpcornil-git** [comment]
Could you reattach the 3 lines as a file or mark them as code (they were cut after "pref_disa>" ? (nothing abnormal up to there)

Next time, before touching anything, could restart the VM to confirm that the issue is not coming from there (not sure I understand then why removing instanaces and adding them again would help but I would like to exclude that possible root cause as well) ?

**Comment by Jlauper** [comment]
I have done some tests. Even on a new VM under Proxmox, the connection drops immediately when Homeassistant is restarted. The webboxes must now be deleted. After a restart of Homeassitant, the boxes can be set up again and work immediately.
The problem does not occur with an installation (no VM) on Raspberry.

**Comment by jpcornil-git** [comment]
This seems to confirm what I mentionned before, i.e. that the host running the VM or the VM doesn't release the comm port (docker for Windows has a similar issue)
